src/test/java/org/opensearch/security/dlic/rest/api/TenantsActionTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.dlic.rest.api;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.apache.hc.core5.http.HttpStatus;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.opensearch.security.test.helper.rest.RestHelper.HttpResponse;
+
+import static org.opensearch.security.OpenSearchSecurityPlugin.PLUGINS_PREFIX;
+
+public class TenantsActionTest extends AbstractRestApiUnitTest {
+    private final String ENDPOINT;
+
+    protected String getEndpointPrefix() {
+        return PLUGINS_PREFIX;
+    }
+
+    public TenantsActionTest() {
+        ENDPOINT = getEndpointPrefix() + "/api/tenants";
+    }
+
+    @Test
+    public void testParallelPutRequests() throws Exception {
+
+        setup();
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        String[] caseOneRequests = { ENDPOINT + "/tenant1", ENDPOINT + "/tenant1" };
+        String[] caseOneBodies = { "{\"description\":\"create tenant 1\"}", "{\"description\":\"create tenant 2\"}" };
+        HttpResponse[] caseOneResponses = executeMultipleAsyncPutRequest(caseOneRequests, caseOneBodies);
+        boolean created = false;
+        for (HttpResponse response : caseOneResponses) {
+            int sc = response.getStatusCode();
+            switch (sc) {
+                case HttpStatus.SC_CREATED:
+                    Assert.assertFalse(created);
+                    created = true;
+                    break;
+                case HttpStatus.SC_OK:
+                    break;
+                default:
+                    Assert.assertEquals(HttpStatus.SC_CONFLICT, sc);
+                    break;
+            }
+        }
+
+        String[] caseTwoRequests = { ENDPOINT + "/tenant1", ENDPOINT + "/tenant1", ENDPOINT + "/tenant1",
```

Comment:
Further tests are unneeded, so long as the 1 PUT is accepted with CREATED, and the 2nd PUT is rejected with CONFLICT this is working as indented.